language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/position/PositionTreeVisitor.java

```diff
@@ -151,17 +116,24 @@ public void visit(BLangImportPackage importPkgNode) {
         acceptNode(pkgEnv.node);
     }
 
-    public void visit(BLangXMLNS xmlnsNode) {
-        // No implementation
-    }
-
     public void visit(BLangFunction funcNode) {
         // Check for native functions
         BSymbol funcSymbol = funcNode.symbol;
         if (Symbols.isNative(funcSymbol)) {
             return;
         }
 
+        if (funcNode.getPosition().sLine == this.position.getLine() && funcNode.getPosition().sCol == 1) {
+            this.context.put(NodeContextKeys.NODE_KEY, funcNode);
```

Comment:
Let's try to have the following context populating code segment to a util within the tree visitor (Just a suggestion :) ).

Reply:
it'll be good if we move this to a util. Let's do a full fetch design and code review on ballerina language server and try to identify what can be reuse and how they can be placed as now we have almost all the important language features .